tracer_options.go

```diff
@@ -28,143 +27,16 @@ func (e *Endpoint) toZipkin() *model.Endpoint {
-	sampler zipkin.Sampler
-
-	// unsampledNoop turns potentially expensive operations on unsampled
-	// Spans into no-ops. More precisely, tags and log events are silently
-	// discarded. If NewSpanEventListener is set, the callbacks will still fire.
-	unsampledNoop *bool
-
-	// clientServerSameSpan allows for Zipkin V1 style span per RPC. This places
-	// both client end and server end of a RPC call into the same span.
-	sharedSpans *bool
-
-	// traceID128Bit enables the generation of 128 bit traceIDs in case the tracer
-	// needs to create a root span. By default regular 64 bit traceIDs are used.
-	// Regardless of this setting, the library will propagate and support both
-	// 64 and 128 bit incoming traces from upstream sources.
-	traceID128Bit *bool
-
-	noop *bool
-
-	defaultTags map[string]string
-
 	observer otobserver.Observer
 }
 
 // TracerOption allows for functional options.
 // See: http://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
-type TracerOption func(opts *TracerOptions) error
-
-// WithLocalEndpoint sets the local endpoint of the tracer.
-func WithLocalEndpoint(e Endpoint) TracerOption {
-	return func(opts *TracerOptions) error {
-		opts.localEndpoint = e.toZipkin()
-		return nil
-	}
-}
-
-// WithSampler allows one to add a Sampler function
-func WithSampler(sampler Sampler) TracerOption {
-	return func(opts *TracerOptions) error {
-		opts.sampler = zipkin.Sampler(sampler)
-		return nil
-	}
-}
-
-// TrimUnsampledSpans option, similar to WithNoopSpan
-// in zipkin options
-func TrimUnsampledSpans(trim bool) TracerOption {
-	return func(opts *TracerOptions) error {
-		opts.unsampledNoop = &trim
-		return nil
-	}
-}
-
-// TraceID128Bit option
-func TraceID128Bit(val bool) TracerOption {
-	return func(opts *TracerOptions) error {
-		opts.traceID128Bit = &val
-		return nil
-	}
-}
-
-// ClientServerSameSpan allows to place client-side and server-side annotations
-// for a RPC call in the same span (Zipkin V1 behavior) or different spans
-// (more in line with other tracing solutions). By default this Tracer
-// uses shared host spans (so client-side and server-side in the same span).
-// If using separate spans you might run into trouble with Zipkin V1 as clock
-// skew issues can't be remedied at Zipkin server side.
-//
-// similar to zipkin.WithSharedSpans
-func ClientServerSameSpan(val bool) TracerOption {
-	return func(opts *TracerOptions) error {
-		opts.sharedSpans = &val
-		return nil
-	}
-}
-
-// WithTags allows one to set default tags to be added to each created span
-func WithTags(tags map[string]string) TracerOption {
-	return func(opts *TracerOptions) error {
-		for k, v := range tags {
-			opts.defaultTags[k] = v
-		}
-		return nil
-	}
-}
-
-// WithNoopTracer allows one to start the Tracer as Noop implementation.
-func WithNoopTracer(tracerNoop bool) TracerOption {
-	return func(opts *TracerOptions) error {
-		opts.noop = &tracerNoop
-		return nil
-	}
-}
+type TracerOption func(opts *TracerOptions)
```

Comment:
Not 100% about this change but in the other hand I don't see this failing in any case, cc @basvanbeek

Reply:
this is fine... there could be other optional OT specific settings we want to add to a wrapped tracer...